core/src/main/java/org/apache/iceberg/util/RocksDBStructLikeMap.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.types.Serializer;
+import org.apache.iceberg.types.Serializers;
+import org.apache.iceberg.types.Types;
+import org.rocksdb.Options;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+import org.rocksdb.WriteOptions;
+
+public class RocksDBStructLikeMap extends AbstractMap<StructLike, StructLike> implements Map<StructLike, StructLike> {
+
+  static {
+    RocksDB.loadLibrary();
+  }
+
+  public static RocksDBStructLikeMap create(String path,
```

Comment:
Currently,  if multiple tasks are located in the same node,  then they will use the same rocksdb location. That will mess up the spilled data. We will need to create separate rocksdb dir for different tasks even if they are located in the same node.

Reply:
Hello Openinx, as different flink tasks are in different processes, it is hard to coordinate. Can we pass a Pathes list to create function, and use a static variable to make sure every operator can round robin between different discs.

And shall we also introduce the rocksdb profile codes in flink? to enable some more compaction methods like fifo, in case people know what are they doing.

Another thinking is shall we put limit on the rocksdb memory usage? in case these tasks are running on yarn, if memory exceeds the memory limit of yarn, the process could be killed.

let me know your thinking :)

Reply:
> We will need to create separate rocksdb dir for different tasks even if they are located in the same node.

I think this comment is out of date now because we've already generated an unique directory name for each `BaskTaskWriter` now , please see here
https://github.com/apache/iceberg/pull/2680/files#diff-e550ee80e8343a3396e67ab7fec4fe50a2ac633b02682cdc330d9a2b719e1a5eR61-R68 . 

Yes,  I think we could add more rocksdb options (such as rocksdb writer buffer size etc) in here so that we could control what's the exactly behavior in rocksdb. https://github.com/apache/iceberg/pull/2680/files#diff-125c2d685d98fcab9fcf124dd51f55dafe83462f6aa0462eeb1fe034e136f8afR86-R91

Thanks for the great feedback @zhougit86 !

Reply:
I think allow changing RocksDB `DBOptions` and `ColumnFamilyOptions` is quite important for debugging / tuning OOM issue with RocksDB. Maybe we should allow user to supply a config factory, similar to how Flink does with its RocksDB state backend.